### PR TITLE
Normative: Create the PluralCategories array each time

### LINKED ISSUE #224: Copy [[PluralCategories]] in Intl.PluralRules.prototype.resolvedOptions?
Follow-up from https://github.com/tc39/proposal-intl-plural-rules/issues/28#issuecomment-341557030

Seems like I've misinterpreted the SpiderMonkey implementation, because SpiderMonkey definitely returns a copy of the `[[PluralCategories]]` array in `Intl.PluralRules.prototype.resolvedOptions()`:  https://searchfox.org/mozilla-central/rev/6e96a3f1e44e286ddae5fdafab737709741d237a/js/src/builtin/intl/PluralRules.js#235-240

### PR DESCRIPTION
**Pull request by littledan**
Closes #224

### CODE REVIEW

File: spec/pluralrules.html [hunk 1]
```diff
@@ -26,8 +26,6 @@ <h1>InitializePluralRules ( _pluralRules_, _locales_, _options_ )</h1>
         1. Let _localeData_ be %PluralRules%.[[LocaleData]].
         1. Let _r_ be ResolveLocale(%PluralRules%.[[AvailableLocales]], _requestedLocales_, _opt_, %PluralRules%.[[RelevantExtensionKeys]], _localeData_).
         1. Set _pluralRules_.[[Locale]] to the value of _r_.[[locale]].
-        1. Let _pluralCategories_ be a List of Strings representing the possible results of <emu-xref href="#sec-pluralruleselect">PluralRuleSelect</emu-xref> for the selected locale _pluralRules_.[[Locale]].
-        1. Set _pluralRules_.[[PluralCategories]] to CreateArrayFromList(_pluralCategories_).
```
[the remark this answers is not in the document]
Comment (littledan) [reply spec/pluralrules.html]: Do you see any advantage to that phrasing?

[the remark this answers is not in the document]
Comment (ljharb) [reply spec/pluralrules.html]: Only that it makes clear that the contents of the array are fixed at initialization time, but the array is newly created every time. Not a huge benefit either way.

**Comment by littledan** [comment]
Anyone interested in writing test262 tests for this change? cc @ms2ger

**Comment by Ms2ger** [comment]
The test for this change seems to have landed already: https://github.com/tc39/test262/pull/1576

**Comment by littledan** [comment]
@caridy Any hesitation to landing this patch, or should we land it?

File: spec/pluralrules.html [hunk 2]
```diff
@@ -355,7 +351,6 @@ <h1>Properties of Intl.PluralRules Instances</h1>
       <li>[[MinimumIntegerDigits]] is a non-negative integer Number value indicating the minimum integer digits to be used.</li>
       <li>[[MinimumFractionDigits]] and [[MaximumFractionDigits]] are non-negative integer Number values indicating the minimum and maximum fraction digits to be used. Numbers will be rounded or padded with trailing zeroes if necessary.</li>
       <li>[[MinimumSignificantDigits]] and [[MaximumSignificantDigits]] are positive integer Number values indicating the minimum and maximum fraction digits to be used. Either none or both of these properties are present; if they are, they override minimum and maximum integer and fraction digits.</li>
-      <li>[[PluralCategories]] is an array of unique string values, from the the list `"zero"`, `"one"`, `"two"`, `"few"`, `"many"` and `"other"`, that are relevant for the locale whose localization is specified in LDML Language Plural Rules.</li>
```
Comment (anba) [inline spec/pluralrules.html]: I think it'd be useful to keep the info about the possible plural category values and that the returned array doesn't contain duplicates somewhere in the spec.

File: spec/pluralrules.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (caridy) [reply spec/pluralrules.html]: I think those are optimizations that engines can do, and we should keep the spec as clean as possible. Creating this every time in the `resolvedOptions` is fine. It isn't a very common operation anyways.

File: spec/pluralrules.html (hunk 2 above)
Comment (caridy) [reply spec/pluralrules.html]: yes, let's keep this around somewhere! Maybe as a note in resolvedOptions.

Comment (littledan) [reply spec/pluralrules.html]: Good point. I've moved this text into resolvedOptions.

**Pull request merged**